packages/calcite-components/src/components/list/list.tsx

```diff
@@ -649,6 +653,10 @@ export class List
     }
   }
 
+  private setListItemGroups(): void {
+    this.listItemGroups = Array.from(this.el.querySelectorAll(listItemGroupSelector));
```

Comment:
The other alternative would be something like this:

```tsx
let ancestor = el.closest("calcite-list-item-group");
const headings: string[] = [];
while (ancestor) {
  headings.push(ancestor.heading);
  ancestor = ancestor.closest("calcite-list-item-group");
}
```

I'm not sure which would be more performant, calling closest() a few times on connectedCallback or calling `setListItemGroups` on each mutation. Thoughts @jcfranco?